java-client/deephaven/src/main/java/io/deephaven/client/impl/DeephavenClientSession.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2016-2021 Deephaven Data Labs and Patent Pending
+ */
+
+package io.deephaven.client.impl;
+
+import io.deephaven.qst.table.TableSpec;
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.ClientInterceptors;
+import io.grpc.ForwardingClientCall;
+import io.grpc.ManagedChannel;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import org.apache.arrow.flight.FlightClient;
+import org.apache.arrow.flight.FlightGrpcUtilsExtension;
+import org.apache.arrow.memory.BufferAllocator;
+
+import java.util.Collections;
+
+public class DeephavenClientSession extends FlightSession implements BarrageSubscription.Factory {
+
+    /**
+     * The server is able to make more efficient use of resources when the same updateInterval is used across all API
+     * clients. Set this default on the server with {@code -Dbarrage.updateInterval}. Otherwise, it is recommended to
+     * sparingly use a value other than the default.
+     */
+    private static final int DEFER_UPDATE_INTERVAL_TO_SERVER = 0;
```

Comment:
I think it *could* be nice if this were part of `BarrageSubscriptionOptions`, and documented there.